src/Compilers/Core/Portable/DiagnosticAnalyzer/DefaultAnalyzerAssemblyLoader.cs

```diff
@@ -57,12 +61,17 @@ protected override string PrepareSatelliteAssemblyToLoad(string assemblyFilePath
         /// </summary>
         /// <param name="windowsShadowPath">A shadow copy path will be created on Windows and this value 
         /// will be the base directory where shadow copy assemblies are stored. </param>
-        internal static IAnalyzerAssemblyLoaderInternal CreateNonLockingLoader(string windowsShadowPath, ImmutableArray<IAnalyzerAssemblyResolver>? externalResolvers = null)
+        internal static IAnalyzerAssemblyLoaderInternal CreateNonLockingLoader(
+#if NET
+            AssemblyLoadContext? loadContext,
+#endif
```

Comment:
I've tried to avoid `#if` in the middle of ctor here. Instead I prefer having a complete separate ctor for `NET` where the `ALC` is a parameter

Reply:
Damnit!  that's how i started.  but i didn't like the full dupe of signatures.  :D
for the compielr side, i will switch to that.

Reply:
ok.  changed to larger ifdefs.